scripts/object_storage_cleaner/storage_cleaner.py

```diff
@@ -0,0 +1,628 @@
+                continue
+
+            # Another key, process the previous key.
+            if current_key is not None:
+                result = self._aggregate_if_logically_deleted(
+                    current_key, current_versions
+                )
+                if result:
+                    yield result
+
+            # Start a new batch for the new key
+            current_key = item.key
+            current_versions = [item]
+
+        # Process the final key
+        if current_key is not None:
+            result = self._aggregate_if_logically_deleted(current_key, current_versions)
+            if result:
+                yield result
+
+    def scan(self) -> None:
+        """Scan and print report of logically deleted objects."""
+
+        self._print_filters()
+
+        def progress(delta: int) -> None:
+            self._print_progress(delta)
+
+        for aggregate, versions in self._iter_logically_deleted(progress=progress):
+            self._stats_keys += 1
+            self._stats_versions += aggregate.versions_count + 1
+            self._stats_bytes += aggregate.total_size_bytes
+
+            if logger.isEnabledFor(TRACE_LEVEL):
+                for v in versions:
+                    logger.log(
+                        TRACE_LEVEL,
+                        "Version: Key=%s VersionId=%s Size=%s LastModified=%s",
+                        v.key,
+                        v.version_id,
+                        v.size,
+                        v.last_modified,
+                    )
+
+        self._print_summary(prune=False)
+
+    def prune(self) -> None:
+        """Scan and delete logically deleted objects."""
+
+        self._print_filters()
+
+        batch: list[dict[str, str]] = []
+
+        def progress(delta: int) -> None:
+            self._print_progress(delta, label="Deleted")
+
+        logger.debug("Starting to delete logically deleted objects...")
+
+        # Iterate
+        for aggregate, versions in self._iter_logically_deleted(progress=progress):
+            self._stats_keys += 1
+            self._stats_bytes += aggregate.total_size_bytes
+
+            for v in versions:
+                batch.append({"Key": v.key, "VersionId": v.version_id})
+                if len(batch) >= self.BATCH_SIZE:
+                    self._stats_versions += self._delete_batch(batch)
+                    batch.clear()
+                    self._print_progress(0, label="Deleted")
+
+        # Flush remaining
+        if batch:
+            self._stats_versions += self._delete_batch(batch)
+
+        self._print_summary(prune=True)
+
+
+def _handle_sigint(sig: int, frame: object) -> None:
+    logger.info("\n\nScan interrupted by user.")
+    sys.exit(0)
+
+
+signal.signal(signal.SIGINT, _handle_sigint)
+
+
+def setup_logging(debug: bool = False, log_file: str | None = None) -> None:
+    """Setup logging for the script."""
+    level = TRACE_LEVEL if log_file else (logging.DEBUG if debug else logging.INFO)
```

Comment:
Please do not use inline `if` in QFieldCloud code. This is unreadable by the common folks. Do not apply such magic, make as many of the configuration required and explicit. The log level, if not by default, should be passed as a cli param/envvar and no magical logic should be applied.